crates/api_common/src/request.rs

```diff
@@ -65,95 +65,70 @@ pub async fn fetch_link_metadata(url: &Url, context: &LemmyContext) -> LemmyResu
   })
 }
 
-/// Generate post thumbnail in background task, because some sites can be very slow to respond.
+/// Generates and saves a post thumbnail and metadata.
 ///
 /// Takes a callback to generate a send activity task, so that post can be federated with metadata.
 ///
 /// TODO: `federated_thumbnail` param can be removed once we federate full metadata and can
 ///       write it to db directly, without calling this function.
 ///       https://github.com/LemmyNet/lemmy/issues/4598
-pub fn generate_post_link_metadata(
+pub async fn generate_post_link_metadata(
   post: Post,
   custom_thumbnail: Option<Url>,
-  federated_thumbnail: Option<Url>,
```

Comment:
Dont remove this, as Lemmy would generate a lot of extra local thumbnails when it could simply reuse the thumbnail of the original instance.

Reply:
The reasons I removed it, had to do with complex cases w/ respect to proxying:

- The `post.url` is the source URL or picture link, it should always remain unproxied / unchanged.
- The `thumbnail_url` could be a pictrs proxied picture, a local picture, or an external picture.
- Federating that `thumbnail_url` could be dangerous, because it *could* be a proxied link already. If we have proxying turned on for our local instance, then the URL would be double-proxied. 

I could add back in `federated_thumbnail` , but we'd have to ignore it if proxying is turned on for our instance.

Reply:
There is nothing "dangerous" about double-proxying an image. In fact its better because it prevents small sites from being overloaded if lots of instances generate thumbnails for it.

Im also not clear why the post.url should always remain unproxied. If its an image its better to request it through the local instance, to reduce bandwidth usage on the original site and avoid ip leaks.

Reply:
> Im also not clear why the post.url should always remain unproxied.

1. its pointless, since images should be using the `thumbnail_url` field anyway. In fixing this, I also ran into a bug where even non-picture link URLs were getting proxied. 
2. if you chain proxy links, its going to get messy really quickly. A picture shouldn't need to get proxied multiple times and put that much load on pictrs servers. Not to mention the speed issues. Why not just leave the post.url alone, and have the *derived* images be generated from it.

I could add back in federated_thumbnail, but ignore it and regenarate it if we have proxying turned on. That's a small one-time cost that doesn't chain and put load and speed issues on a lot of servers.

Reply:
Alright lets leave it like this and maybe adjust things after its in production.